### PR TITLE
Resolvers timeouts plugin: ability to specify timeouts per field resolver.

### PR DESCRIPTION
**Pull request by auvn**

### CODE REVIEW

File: plugin/timeout/wrappers.gotpl [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{{ reserveImport "context"  }}
+{{ reserveImport "time"  }}
+
+{{ range $w := .Wrappers }}
+    type {{$w.Type}} struct {
+        cfg TimeoutsConfig
+        parent {{$w.ObjectType}}
+    }
+
+    func New{{$w.Type}}(parent {{$w.ObjectType}}, cfg TimeoutsConfig) *{{$w.Type}}{
```
Comment (unkeep) [inline plugin/timeout/wrappers.gotpl:10]: what about to generate a method of the original resolver type, which wraps the receiver?
```go
NewOriginalResolver(...).WithTimeouts(cfg)
```

Comment (unkeep) [reply plugin/timeout/wrappers.gotpl:10]: or `WithDecorators(...)` which will apply all available decorators

Comment (auvn) [reply plugin/timeout/wrappers.gotpl]: added decorators generation

File: plugin/resolver/timeout/wrappers!.gotpl [hunk 2]
```diff
@@ -2,32 +2,34 @@
 {{ reserveImport "time"  }}
```
Comment (mkorolyov) [inline plugin/resolver/timeout/wrappers!.gotpl:2]: file name became `wrappers!.gotpl`

**Review by mkorolyov — CHANGES_REQUESTED** [review]

File: plugin/resolver/timeout/wrappers!.gotpl (hunk 2 above)
Comment (auvn) [reply plugin/resolver/timeout/wrappers!.gotpl:2]: it's fine !.gotpl templates goes first in the graphql template rending engine

``` go
		// important files go first
		if strings.HasSuffix(roots[i], "!.gotpl") {
			return true
		}

```

Comment (auvn) [reply plugin/resolver/timeout/wrappers!.gotpl:2]: up

**Review by mkorolyov — APPROVED** [review]

**Pull request merged**